src/libraries/Microsoft.PowerFx.Interpreter/Environment/PowerFxConfigExtensions.cs

```diff
@@ -34,16 +34,31 @@ public static void EnableSetFunction(this PowerFxConfig powerFxConfig)
         /// Enable all multation functions which allows scripts to execute side effect behavior.
         /// </summary>
         /// <param name="symbolTable"></param>
+        [Obsolete("This will be removed. Enable mutation functions by calling PowerFxConfig.EnableMutationFunctions()")]
         public static void EnableMutationFunctions(this SymbolTable symbolTable)
         {
             symbolTable.AddFunction(new RecalcEngineSetFunction());
-            symbolTable.AddFunction(new CollectFunction());
             symbolTable.AddFunction(new PatchFunction());
             symbolTable.AddFunction(new RemoveFunction());
             symbolTable.AddFunction(new ClearFunction());
             symbolTable.AddFunction(new ClearCollectFunction());
         }
 
+        public static void EnableMutationFunctions(this PowerFxConfig config)
+        {
+            config.SymbolTable.AddFunction(new RecalcEngineSetFunction());
+            config.SymbolTable.AddFunction(new PatchFunction());
+            config.SymbolTable.AddFunction(new RemoveFunction());
+            config.SymbolTable.AddFunction(new ClearFunction());
+            config.SymbolTable.AddFunction(new ClearCollectFunction());
+
+            foreach (KeyValuePair<TexlFunction, IAsyncTexlFunction> func in Library.MutationFunctions())
+            {
+                config.SymbolTable.AddFunction(func.Key);
+                config.AdditionalFunctions.Add(func.Key, func.Value);
```

Comment:
>config.AdditionalFunctions. [](http://example.com/codeflow?start=16&length=27)

I guess this is the tricky part ... 
how to associate the TexlFunction with the new implementation.

Reply:
See EnableJsonFunctions()  as an alternative pattern. 

That would let us keep this on symbol table.

Reply:
Yes. That's why I opted to follow this path. I tried other approaches but then I ended up changing some core objects like the parser.